composer.json

```diff
@@ -57,7 +57,7 @@
     ]
   },
   "archive": {
-    "exclude": ["Jenkinsfile", "Dockerfile", ".DS_STORE", ".idea", ".phan", ".docker", "ruleset.xml", "phan.*", ".gitignore", "build.xml", ".github", "supervisord.conf", "entrypoint.sh", "/magento"]
+    "exclude": ["Jenkinsfile", "Dockerfile", ".DS_STORE", ".idea", ".phan", ".docker", "composer.lock", "ruleset.xml", "phan.*", ".gitignore", "build.xml", ".github", "supervisord.conf", "entrypoint.sh", "default.conf",  "/magento"]
```

Comment:
We can't really exclude the `composer.lock` file from the archive. If the composer is run on the directory, will pull the wrong package versions.

Reply:
@supercid ,  i think it`s possible to fix by right requirements in composer.json for require-dev, if composer run is used on CI and self-testing of code.

Reply:
The case I was thinking was if someone tries to update the extension later on by using composer instead of a marketplace package.
Without the lock file, a `composer update` could pull dependencies that we haven't tested yet, thus breaking the installation. I think it's safer to leave there.

Reply:
If it`s about "composer update" for Magento installation and  udpate nosto/module-nostotagging - Composer only ever installs the packages listed as "require-dev" of your master composer.json file, and if these packages do need something else, then only their "require" packages are installed, but not their "require-dev" packages.

Reply:
The main reason we want to keep `composer.lock` in the archive  (and in the repo) is the fact that our PHP SDK (https://github.com/Nosto/nosto-magento2/blob/develop/composer.json#L40) is a dependency in many of our projects and the SDK is developed somewhat separately. We basically want to lock down the SDK version for each M2 module release. 

Is the composer.lock and defaults.conf (used by our internal CI) causing some problems for you @vmalyk?

Reply:
@hannunosto thank you for review,

Firstly, issue is related with 2 extra files in Composer packages from Magento Marketplace and these 2 files have no sense for real Magento installation, because Nosto module is installed with "require" modules and no require require-dev (and It`s right behavior of Composer).  Of course, you can stay use these  files in repository and use in CI. 
Secondly, if you want lock the SDK for release you`ll do that in require. Example how composer.lock doesn`t work for  installations:
*Pre-conditions*
1. In composer require you have  "nosto/php-sdk": "^4.0.5"